desc/objectives/_coils.py

```diff
@@ -708,6 +716,392 @@ def compute(self, params, constants=None):
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        coilset = self.things[0]
+        grid = self._grid or LinearGrid(N=16)
+
+        self._dim_f = coilset.num_coils
+        self._constants = {"coilset": coilset, "grid": grid, "quad_weights": 1.0}
+
+        if self._normalize:
+            coils = tree_leaves(coilset, is_leaf=lambda x: not hasattr(x, "__len__"))
+            scales = [compute_scaling_factors(coil)["a"] for coil in coils]
+            self._normalization = np.mean(scales)  # mean length of coils
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute minimum distances between coils.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of coilset degrees of freedom, eg CoilSet.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc.
+            Defaults to self._constants.
+
+        Returns
+        -------
+        f : array of floats
+            Minimum distance to another coil for each coil in the coilset.
+
+        """
+        if constants is None:
+            constants = self.constants
+        pts = constants["coilset"]._compute_position(
+            params=params, grid=constants["grid"]
+        )
+
+        def body(k):
+            # dist btwn all pts; shape(ncoils,num_nodes,num_nodes)
+            dist = safenorm(pts[k][None, :, None] - pts[:, None, :], axis=-1)
+            # exclude distances between points on the same coil
+            mask = jnp.ones(self.dim_f).at[k].set(0)[:, None, None]
+            return jnp.min(dist, where=mask, initial=jnp.inf)
+
+        min_dist_per_coil = fori_loop(
+            0,
+            self.dim_f,
+            lambda k, min_dist: min_dist.at[k].set(body(k)),
+            jnp.zeros(self.dim_f),
+        )
+        return min_dist_per_coil
+
+
+class PlasmaCoilsetMinDistance(_Objective):
+    """Target the minimum distance between the plasma and coilset.
+
+    Will yield one value per coil in the coilset, which is the minimumm distance from
+    that coil to the plasma boundary surface.
+
+    NOTE: By default, assumes the plasma boundary is not fixed and its coordinates are
+    computed at every iteration, for example if the equilibrium is changing in a
+    single-stage optimization.
+    If the plasma boundary is fixed, set eq_fixed=True to precompute the last closed
+    flux surface coordinates and improve the efficiency of the calculation.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
```

Comment:
Should we edit this to mention that it also works for any `Surface` in addition to an `Equilibrium`? That is also missing from the documentation of `PlasmaVesselDistance`

Reply:
Yes but `PlasmaVesselDistance` only works for an eq and a FourieRRZToroidalSurface. I don't know of a situation where you have two surfaces you care about (maybe blanket and vessel I guess...) but that is not going to be something we change in this PR, we can make an issue for it if needed